src/rdkafka.h

```diff
@@ -4205,6 +4296,987 @@ rd_kafka_interceptor_add_on_request_sent (
+ *         in the cluster.
+ *
+ * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success, or an error code
+ *          if the arguments were invalid.
+ *
+ * @sa rd_kafka_AdminOptions_set_validate_only()
+ */
+RD_EXPORT rd_kafka_resp_err_t
+rd_kafka_NewTopic_set_replica_assignment (rd_kafka_NewTopic_t *new_topic,
+                                          int32_t partition,
+                                          int32_t *broker_ids,
+                                          size_t broker_id_cnt,
+                                          char *errstr, size_t errstr_size);
+
+/**
+ * @brief Add a (broker-side) topic configuration name/value pair to
+ *        the NewTopic object.
+ *
+ * @remark The name and value are not validated by the client, the validation
+ *         takes place on the broker.
+ *
+ * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success, or an error code
+ *          if the arguments were invalid.
+ *
+ * @sa rd_kafka_AdminOptions_set_validate_only()
+ * @sa http://kafka.apache.org/documentation.html#topicconfigs
+ */
+RD_EXPORT rd_kafka_resp_err_t
+rd_kafka_NewTopic_add_config (rd_kafka_NewTopic_t *new_topic,
+                              const char *name, const char *value);
+
+
+/**
+ * @brief Create topics in cluster as specified by the \p new_topics
+ *        array of size \p new_topic_cnt elements.
+ *
+ * @param new_topics Array of new topics to create.
+ * @param new_topic_cnt Number of elements in \p new_topics array.
+ * @param options Optional admin options, or NULL for defaults.
+ * @param rkqu Queue to emit result on.
+ *
+ * Supported admin options:
+ *  - rd_kafka_AdminOptions_set_validate_only() - default false
+ *  - rd_kafka_AdminOptions_set_operation_timeout() - default 0
+ *  - rd_kafka_AdminOptions_set_timeout() - default socket.timeout.ms
+ *
+ * @remark The result event type emitted on the supplied queue is of type
+ *         \c RD_KAFKA_EVENT_CREATETOPICS_RESULT
+ */
+RD_EXPORT void
+rd_kafka_CreateTopics (rd_kafka_t *rk,
+                       rd_kafka_NewTopic_t **new_topics,
+                       size_t new_topic_cnt,
+                       const rd_kafka_AdminOptions_t *options,
+                       rd_kafka_queue_t *rkqu);
+
+
+/**
+ * @brief CreateTopics result type and methods
+ */
+
+/**
+ * @returns the request-level error/success for a CreateTopics request.
+ *          Even if the returned value is RD_KAFKA_RESP_ERR_NO_ERROR there
+ *          may be individual topics that failed, extract per-topic information
+ *          with rd_kafka_CreateTopics_result_topics() to check per-topic
+ *          errors.
+ *
+ * @param errstrp is set to a human-readable error string, if there was an
+ *        error.
+ *
+ * @remark This is equivalent and interchangable to rd_kafka_event_error() and
+ *         rd_kafka_event_error_string() on a RD_KAFKA_EVENT_CREATETOPICS_RESULT event.
+ */
+RD_EXPORT rd_kafka_resp_err_t
+rd_kafka_CreateTopics_result_error (
```

Comment:
over-engineered? 
will these separate methods to get the error per type ever add value over just one?

Reply:
oh, you say that in a remark.

I think it's not a good idea to provide these in addition to rd_kafka_event_error because you leave people wondering what they are for - their existence implies there is likely something more to it than a simple wrapper, and means I need to go investigate (spend extra time) before using.

Reply:
You are right, away they go!